internal/literals/numbers.go

```diff
@@ -0,0 +1,277 @@
+			// this guards against the case where the ast.BasicLit is inside an ast.UnaryExpr
+			// and the BasicLit gets evaluated before the UnaryExpr
+			if _, ok := cursor.Parent().(*ast.UnaryExpr); ok {
+				return nil
+			}
+		}
+
+	default:
+		return errors.New("Wrong node Type")
+	}
+
+	strValue := sign + basic.Value
+	// Literals can have underscores for better readabilty
+	strValue = strings.ReplaceAll(strValue, "_", "")
+
+	switch typeInfo {
+	case types.Typ[types.Float32]:
+		fV, err := strconv.ParseFloat(strValue, 32)
+		if err != nil {
+			return err
+		}
+		call = obfuscateFloat32(float32(fV))
+	case types.Typ[types.Float64], types.Typ[types.UntypedFloat]:
+		fV, err := strconv.ParseFloat(strValue, 64)
+		if err != nil {
+			return err
+		}
+		call = obfuscateFloat64(fV)
+	}
+
+	if call != nil {
+		cursor.Replace(call)
+		return nil
+	}
+
+	// Explicitly typed integers can have a decimal place
+	splitStrValue := strings.Split(strValue, ".")
+	intValue, err := strconv.Atoi(splitStrValue[0])
+	if err != nil {
+		return err
+	}
+
+	switch typeInfo {
+	case types.Typ[types.Int], types.Typ[types.UntypedInt]:
+		call = obfuscateInt(intValue)
+	case types.Typ[types.Int8]:
+		call = obfuscateInt8(int8(intValue))
+	case types.Typ[types.Int16]:
+		call = obfuscateInt16(int16(intValue))
+	case types.Typ[types.Int32]:
+		call = obfuscateInt32(int32(intValue))
+	case types.Typ[types.Int64]:
+		call = obfuscateInt64(int64(intValue))
+	case types.Typ[types.Uint]:
+		call = obfuscateUint(uint(intValue))
+	case types.Typ[types.Uint8]:
+		call = obfuscateUint8(uint8(intValue))
+	case types.Typ[types.Uint16]:
+		call = obfuscateUint16(uint16(intValue))
+	case types.Typ[types.Uint32]:
+		call = obfuscateUint32(uint32(intValue))
+	case types.Typ[types.Uint64]:
+		call = obfuscateUint64(uint64(intValue))
+	case types.Typ[types.Uintptr]:
+		call = obfuscateUintptr(uintptr(intValue))
+	}
+
+	if call == nil {
+		return errors.New("Node is not Integer")
+	}
+
+	cursor.Replace(call)
+	return nil
+}
+
+func bytesToUint(length int, bits string) ast.Expr {
+	var expr ast.Expr
+	for i := 0; i < length; i++ {
+		posStr := strconv.Itoa(i)
+		switch i {
+		case 0:
+			expr = callExpr(ident("uint"+bits), indexExpr("data", intLiteral(posStr)))
+		default:
+			shiftValue := strconv.Itoa(i * 8)
+			expr = &ast.BinaryExpr{
+				X:  expr,
+				Op: token.OR,
+				Y: &ast.BinaryExpr{
+					X:  callExpr(ident("uint"+bits), indexExpr("data", intLiteral(posStr))),
+					Op: token.SHL,
+					Y:  intLiteral(shiftValue),
+				},
+			}
+		}
+	}
+
+	return expr
+}
+
+func obfuscateUint8(data uint8) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	length := 1
+	b := []byte{byte(data)}
+
+	block := obfuscator.obfuscate(b)
+	convertExpr := bytesToUint(length, "8")
+
+	block.List = append(block.List, returnStmt(convertExpr))
+
+	return lambdaCall(ident("uint8"), block)
+}
+
+func obfuscateUint16(data uint16) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	length := 2
```

Comment:
Comment on all statically defined lengths.

I suggest to get all sizes in a more correct way through `types.SizesFor("compiler", "arch").Sizeof(...)`